spec/civildate.html

```diff
@@ -1,164 +1,566 @@
+          1. Let _y_ be _date_.[[Year]].
+        1. If _duration_.[[Month]] is not *undefined*, then
+          1. Let _y_ be _date_.[[Month]] - _duration_.[[Month]].
+        1. Else
+          1. Let _y_ be _date_.[[Month]].
+        1. If _duration_.[[Day]] is not *undefined*, then
+          1. Let _y_ be _date_.[[Day]] - _duration_.[[Day]].
+        1. Else
+          1. Let _y_ be _date_.[[Day]].
+        1. Let _result_ be ? RegulateDate(_y_, _m_, _d_, _disambiguation_).
+        1. Return ? CreateDate(result.[[Year]], result.[[Month]], result.[[Day]]).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause>
+      <h1>Temporal.Date.prototype.with ( _otherDate_, _disambiguation_ )</h1>
+      <p>
+        The `with` method takes one argument _otherDate_.
+        The following steps are taken:
+      </p>
+      <emu-alg>
+        1. Let _date_ be the *this* value.
+        1. If Type(_date_) is not Object or _date_ does not have an [[InitializedTemporalDate]] internal slot, then
+          1. Throw a *TypeError* exception.
+        1. Let _otherDate_ be ? ToPartialDate(_otherDate_).
+        1. Let _disambiguation_ be ? ToString(_disambiguation_).
+        1. If _otherDate_.[[Year]] is not *undefined*, then
+          1. Let _y_ be _otherDate_.[[Year]].
+        1. Else
+          1. Let _y_ be _date_.[[Year]].
+        1. If _otherDate_.[[Month]] is not *undefined*, then
+          1. Let _m_ be _otherDate_.[[Month]].
+        1. Else
+          1. Let _m_ be _date_.[[Month]].
+        1. If _otherDate_.[[Day]] is not *undefined*, then
+          1. Let _d_ be _otherDate_.[[Day]].
+        1. Else
+          1. Let _d_ be _date_.[[Day]].
+        1. Let _result_ be ? RegulateDate(_y_, _m_, _d_, _disambiguation_).
+        1. Return ? CreateDate(result.[[Year]], result.[[Month]], result.[[Day]]).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause>
+      <h1>Temporal.Date.prototype.difference ( _otherDate_ )</h1>
+      <p>
+        The `difference` method takes one argument _otherDate_.
+        The following steps are taken:
+      </p>
+      <emu-alg>
+        1. Let _date_ be the *this* value.
+        1. If Type(_date_) is not Object or _date_ does not have an [[InitializedTemporalDate]] internal slot, then
+          1. Throw a *TypeError* exception.
+        1. Let _otherDate_ be ? ToDate(_otherDate_).
+        1. Let _disambiguation_ be ? ToString(_disambiguation_).
+        1. If ! ValidateDate(_otherDate_.[[Year]], _otherDate_.[[Month]], _otherDate_.[[Day]]) is *false*, then
+          1. Throw a *TypeError* exception.
+        1. <mark>TODO: Throw if any time properties are present?</mark>
+        1. Let _greater_ be _date_.
+        1. Let _smaller_ be _otherDate_.
+        1. <mark>TODO: Swap order if needed.</mark>
+        1. Let _years_ be _greater_.[[Year]] - _smaller_.[[Year]].
+        1. Let _months_ be _greater_.[[Month]] - _smaller_.[[Month]].
+        1. Let _days_ be _greater_.[[Day]] - _smaller_.[[Day]].
```

Comment:
This is wrong (at least in current thinking) because the month introduces the opportunity to have unbalances in addition.
So `date.difference` does not ever result in a duration with a non-zero `.months` element. Instead we go for the number of days delta. That way we can ensure addability so that smaller.plus(duration) == larger and larger.minus(duration) == smaller.